Fix inconsistent vertical offset in Prefs Window panels: remove unnec…

svobs opened the pull request:
- [ ] This change has been discussed with the author.
- [x] It implements / fixes issue #3717.

---

**Description:**
Fix inconsistent vertical offset in Prefs Window panels: removed an unnecessary and unreliable scroll(.zero) request. This resulted in a consistent vertical offset which was too large and easily reduced. The pref panes should always be fixed size and not scrollable, because their superview is an NSScrollView which will provide the scroll. Also, they do not need an explicit scroll() because their superview will always have an implicit scroll of zero when they are redrawn in it.

low-batt reviewed (approved):
I reproduced the bad behavior with a build of the develop branch. I fetched the PR locally and confirmed the changes correct the issue with the vertical offset. Changes look good to me.

lhc70000 commented:
This is an old problem that we do want to fix.

However, I think the `preferenceContentIsScrollable` property was designed for the key bindings pane - we want the table always fill the whole super view and resize with the window. This PR breaks the key bindings pane's layout. Is it possible to keep the current behavior but only remove the extra space at the top?

svobs commented:
> However, I think the preferenceContentIsScrollable property was designed for the key bindings pane - we want the table always fill the whole super view and resize with the window. This PR breaks the key bindings pane's layout. Is it possible to keep the current behavior but only remove the extra space at the top?

Gotcha. I see that now...

I'll try to find time this weekend to see what can be done.

svobs commented:
I think I fixed it. I ended up leaving in the `preferenceContentIsScrollable` logic but took out the problematic calls to change `scrollView`, and just adjusted the vertical offsets for a couple of constraints.

I also noticed that the Advanced page could benefit from having its table expand down with the window as well, so figured better to beg forgiveness than ask for permission? 😇 It's the single reason why `PrefAdvancedViewController.swift` has changes.
